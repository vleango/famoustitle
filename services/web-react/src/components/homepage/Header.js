import React from 'react';

export default (props) => {
    return (
        <header className="pt-5">
            <div className="container">
                <div className="row">
                    <div className="col-lg-12">
                        <h2>famoustitle! (beta)</h2>
                        <p className="subtitle text-muted">
                            News articles with helpful tips and how-to guides for all things
                            related to software development. The main focus of these articles
                            is to find better approaches to the software development process.
                        </p>
                    </div>
                </div>
            </div>
        </header>
    );
}
