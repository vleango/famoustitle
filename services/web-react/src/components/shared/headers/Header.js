import React from 'react';
import { Link } from 'react-router-dom';
import { Breadcrumb, BreadcrumbItem } from 'reactstrap';

export default class Header extends React.Component {

    render() {
        return (
            <div>
                <Breadcrumb tag="nav">
                    <BreadcrumbItem tag={Link} to="/">Home</BreadcrumbItem>
                    <BreadcrumbItem active tag="span">{this.props.resourceTitle}</BreadcrumbItem>
                </Breadcrumb>
            </div>
        );
    }

}
