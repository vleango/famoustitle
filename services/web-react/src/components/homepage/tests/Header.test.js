import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('Components', () => {
  describe('HomePage', () => {
    describe('Header', () => {

      let wrapper = shallow(<Header />);

      describe('Snapshot', () => {
        it('should correctly render Header', () => {
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
});
