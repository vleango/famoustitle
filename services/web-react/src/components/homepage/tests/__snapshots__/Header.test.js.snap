// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components HomePage Header Snapshot should correctly render Header 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        <div
                className="row"
        >
                <div
                        className="col-lg-12"
                >
                        <h2>
                                famoustitle! (beta)
                        </h2>
                        <p
                                className="subtitle text-muted"
                        >
                                News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
                        </p>
                </div>
        </div>
</div>,
      "className": "pt-5",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-lg-12"
          >
                    <h2>
                              famoustitle! (beta)
                    </h2>
                    <p
                              className="subtitle text-muted"
                    >
                              News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
                    </p>
          </div>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-lg-12"
>
            <h2>
                        famoustitle! (beta)
            </h2>
            <p
                        className="subtitle text-muted"
            >
                        News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
            </p>
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                famoustitle! (beta)
</h2>,
              <p
                className="subtitle text-muted"
>
                News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
</p>,
            ],
            "className": "col-lg-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "famoustitle! (beta)",
              },
              "ref": null,
              "rendered": "famoustitle! (beta)",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.",
                "className": "subtitle text-muted",
              },
              "ref": null,
              "rendered": "News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <div
                    className="row"
          >
                    <div
                              className="col-lg-12"
                    >
                              <h2>
                                        famoustitle! (beta)
                              </h2>
                              <p
                                        className="subtitle text-muted"
                              >
                                        News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
                              </p>
                    </div>
          </div>
</div>,
        "className": "pt-5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <div
                        className="col-lg-12"
            >
                        <h2>
                                    famoustitle! (beta)
                        </h2>
                        <p
                                    className="subtitle text-muted"
                        >
                                    News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
                        </p>
            </div>
</div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-lg-12"
>
              <h2>
                            famoustitle! (beta)
              </h2>
              <p
                            className="subtitle text-muted"
              >
                            News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
              </p>
</div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  famoustitle! (beta)
</h2>,
                <p
                  className="subtitle text-muted"
>
                  News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.
</p>,
              ],
              "className": "col-lg-12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "famoustitle! (beta)",
                },
                "ref": null,
                "rendered": "famoustitle! (beta)",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.",
                  "className": "subtitle text-muted",
                },
                "ref": null,
                "rendered": "News articles with helpful tips and how-to guides for all things related to software development. The main focus of these articles is to find better approaches to the software development process.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
